app/controllers/doorkeeper/authorizations_controller.rb

```diff
@@ -49,10 +50,21 @@ def render_error
       end
     end
 
+    def can_authorize_response?
+      matching_token? && pre_auth.client.application.confidential? && Doorkeeper.config.custom_access_token_attributes.empty?
+    end
+
+    def prepare_custom_attributes
+      return if Doorkeeper.config.custom_access_token_attributes.empty?
+      return if (token = matching_token?).blank?
+      custom_attributes = token.attributes.symbolize_keys.slice(*Doorkeeper.config.custom_access_token_attributes)
+      pre_auth.custom_access_token_attributes = custom_attributes
+    end
+
     # Active access token issued for the same client and resource owner with
     # the same set of the scopes exists?
     def matching_token?
-      Doorkeeper.config.access_token_model.matching_token_for(
+      @matching_token ||= Doorkeeper.config.access_token_model.matching_token_for(
```

Comment:
As said [here](https://github.com/doorkeeper-gem/doorkeeper/issues/1665), `matching_token_for` should consider custom attributes if they are provided.

Reply:
I've set it up to ignore matching on the custom attributes here - like my comment in the code says, since we're in the pre-auth here the custom attributes haven't been provided yet, so there's nothing to match on. This could find a matching token then, but because `#can_authorize_response?` always returns false if custom attributes are used, it will just copy that tokens custom attributes into the pre-auth, instead of copying them and generating a new grant/token.

Reply:
I suppose someone could send `/oauth/authorize?tenant_id=TENANT_ID&...`, but I'm not sure it's part of the standard.

I don't mind not handling it.